kong/resty/dns/client.lua

```diff
@@ -766,6 +765,8 @@ local function executeQuery(premature, item)
   item.semaphore:post(math_max(item.semaphore:count() * -1, 1))
   item.semaphore = nil
   ngx.sleep(0)
+  -- 3) destroy the resolver
+  r:destroy()
```

Comment:
Let's add some comments to explain the context related to this function, or at least to show that this function comes from some patch.